app/src/test/java/com/kickstarter/viewmodels/ProjectViewModelTest.kt

```diff
@@ -676,43 +680,92 @@ class ProjectViewModelTest : KSRobolectricTestCase() {
     }
 
     @Test
-    fun testRewardsToolbarTitle() {
-        val project = ProjectFactory.project()
+    fun testPledgeToolbarNavigationIcon_whenNativeCheckoutDisabled() {
         setUpEnvironment(environment())
 
-        this.vm.intent(Intent().putExtra(IntentKey.PROJECT, project))
+        this.vm.intent(Intent().putExtra(IntentKey.PROJECT, ProjectFactory.project()))
 
-        this.rewardsToolbarTitle.assertNoValues()
+        this.pledgeToolbarNavigationIcon.assertNoValues()
+    }
 
+    @Test
+    fun testPledgeToolbarNavigationIcon_whenNativeCheckoutEnabled() {
         setUpEnvironment(environmentWithNativeCheckoutEnabled())
 
-        this.vm.intent(Intent().putExtra(IntentKey.PROJECT, project))
+        this.vm.intent(Intent().putExtra(IntentKey.PROJECT, ProjectFactory.project()))
+
+        this.pledgeToolbarNavigationIcon.assertValue(R.drawable.ic_arrow_down)
+
+        this.vm.inputs.fragmentStackCount(1)
+
+        this.pledgeToolbarNavigationIcon.assertValues(R.drawable.ic_arrow_down, R.drawable.ic_arrow_back)
+
+        this.vm.inputs.fragmentStackCount(0)
+
+        this.pledgeToolbarNavigationIcon.assertValues(R.drawable.ic_arrow_down, R.drawable.ic_arrow_back, R.drawable.ic_arrow_down)
```

Comment:
Splitting up these tests makes the business logic much easier to review, thanks for doing that! 🙌